data/test/scenarios/test_role_types.cfg

```diff
@@ -1,3 +1,14 @@
+#####
+# API(s) being tested: [role],[recall]role=
+##
+# Actions:
+# Spawn some units into the recall list.
+# Assign those unit types to a role.
+# Recall using the role.
+##
+# Expected end state:
+# The first unit listed in the role is recalled.
```

Comment:
This isn't [how roles work](https://wiki.wesnoth.org/InternalActionsWML#.5Brole.5D), all of the docs in this file need a rewrite. The `[role]` tag chooses one specific unit at the time the `[role]` tag is read.

Suggestion for this test:
```
#####
# API(s) being tested: [role],[recall]role=
##
# Actions:
# Spawn some units into the recall list.
# Assign a role, choosing a unit based on the unit type.
# Recall using the role.
##
# Expected end state:
# The role has been assigned to the unit whose type appeared first on the list.
# That unit is recalled.
#####
```

Reply:
Alright, will update. Given the text on the wiki in part says:
> However, there is no guarantee that roles will ever be assigned.

I interpreted that to mean the role was evaluated at time of use rather than when it's defined.

Reply:
There's 5 tests in this file, the other 4 still need a similar update.

Reply:
I believe they've been updated?

Reply:
I've added comments to the individual tests, so marking this thread as resolved.